mlir/lib/Util/Dependency.cpp

```diff
@@ -1926,9 +1934,18 @@ void dependencyCanonicalizer::redoDepTraceIfDepOnHier(func::FuncOp func) {
     SmallVector<Value, 1> sink_op_scalar_ins;
     SmallVector<Value, 1> sink_op_scalar_outs;
     // auto &bb = exec_op.getBody().front();
-    Operation &child_op = exec_op.getBody().front().getOperations().front();
+    // Operation &child_op = exec_op.getBody().front().getOperations().front();
     // Operation &child_op = exec_op.getOps().begin();
-    depTracer.getPartialMemrefFromOp(&child_op, sink_op_memref_reads,
+    Operation *child_op = nullptr;
+    // Pick the first op that is not a shape altering op.
+    for (auto &op : exec_op.getBody().front().getOperations()) {
```

Comment:
Nit: air::ExecuteOp now has a method getChildOp() which will get the child op for you.

Reply:
Yes, but this is looking for the first child that is not a shape altering op, and not just the first child op. So I think the loop and checks are still necessary.

Reply:
Ok. I think it's good to merge once CI passes.